apps/src/p5lab/layoutUtils.js

```diff
@@ -0,0 +1,183 @@
+/**
+ * Given a group of sprites, arrange all sprites of this type in a particular
+ * layout. This is likely to change some or all of position/rotation/scale for
+ * the sprites in the group.
+ * Entirely derived from the logic for similar blocks in our dance party repo here (more layouts supported there):
```

Comment:
Would it make sense to reuse some of that logic rather than copy it over here? Perhaps making a common layoutUtils file?

Reply:
Then it may be easier to apply code updates such as those suggested below.

Reply:
The old logic is on the dance repo, so it does feel like the refactor we all want would greatly increase the scope of this task. Given that the curriculum team needs these blocks in order to start building new lessons, I'm hesitant to do that right away. I do think it's a valuable next step. @bencodeorg has already laid some good groundwork for this by creating this file. Next steps would be to:
* copy all other layout logic from api in the dance-party repo into the main repo
* add lab-specific logic for how sprites get resized (see this comment: https://github.com/code-dot-org/code-dot-org/pull/51061#issuecomment-1497490196)
* somehow get dance to use the new utils instead of what's in the old api
* remove the old logic from the api

Meta-question about this: Much of Dance Party's inner workings seem to be self-contained within its own repo. I don't know if anyone besides us actually uses it as a plugin, but would it be bad to remove this logic from the API and instead make it rely on the cdo repo?

Reply:
+1 to Mike's comments -- I think a refactor is a great goal, but seems out of scope given time pressure on curriculum side?

Reply:
Additionally, as we've refined these functions for Sprite Lab (e.g. using a helper to set scale, not updating depth or rotation), we've increased the difference between versions. This doesn't make it impossible to have both labs share utilities, but it does mean we'd have to be more careful. And I definitely wouldn't want to proceed until we had time to discuss as a team which repository the logic belongs in and why.